docs/source/advanced/advanced_gpu.rst

```diff
@@ -73,6 +73,106 @@ Sharded Training can work across all DDP variants by adding the additional ``--p
 
 Internally we re-initialize your optimizers and shard them across your machines and processes. We handle all communication using PyTorch distributed, so no code changes are required.
 
+----------
+
+.. _fully-sharded:
+
+Fully Sharded Training
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. warning::
+    Fully Sharded Training is in beta and the API is subject to change. Please create an `issue <https://github.com/PyTorchLightning/pytorch-lightning/issues>`_ if you run into any issues.
+
+`Fully Sharded <https://fairscale.readthedocs.io/en/latest/api/nn/fsdp.html>`__ shards optimizer state, gradients and parameters across data parallel workers. This allows you to fit much larger models onto multiple GPUs into memory.
+
+Fully Sharded Training alleviates the need to worry about balancing layers onto specific devices using some form of pipe parallelism, and optimizes for distributed communication with minimal effort.
+
+By default, Fully Sharded acts similar to :ref:`sharded` which shards optimizer states and gradients. If you can train with default Fully Sharded, it is recommended to just use :ref:`sharded`.
```

Comment:
@SeanNaren , which arg is controlling FSDP to be just sharing optimizer states and gradients?

Reply:
Ah this is a mistake, by default we do shard activations as well, but if we set `reshard_after_forward=False` then we'll use ZeRO 2 (gradients/optimizers only) from what I've gathered in the [FSDP](https://fairscale.readthedocs.io/en/stable/api/nn/fsdp_tips.html) docs

Reply:
got it. let us fix this then.